### PR TITLE
Owls 103828 - Fix for the Cluster and Domain 'Available' conditions for admin only domains and intentionally shutdown clusters

### PR DESCRIPTION
**Pull request by ankedia**
Owls 103828 - Fix for the Cluster and Domain 'Available' Conditions for admin only domains and intentionally shutdown clusters. In addition, it contains fix for Completed:false for domain with NEVER policy. This PR includes changes from the PR # 3597 ([OWLS-103825](https://github.com/oracle/weblogic-kubernetes-operator/tree/OWLS-103825) branch).
- Set Cluster 'Available' Condition to false if no servers expected to run.
- Set Domain 'Available' Condition to true if only admin server is running and cluster(s) and managed servers are intentionally shutdown.
- Set Domain `Completed` Condition to true if domain start policy is NEVER.

Integration test run - https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/13685/console

### CODE REVIEW

**Comment by alai8** [comment]
@ankedia @lennyphan Does this PR include all the changes from PR 3597 for OWLS-103825? Do we still need PR 3597?

**Comment by lennyphan** [comment]
> @ankedia @lennyphan Does this PR include all the changes from PR 3597 for OWLS-103825? Do we still need PR 3597?

This PR includes the changes from PR 3597.  We made a conflicting change so decided it was best to resolve both JIRA's in one PR, if possible.

**Comment by tbarnes-us** [comment]
Please also update the doc to something similar to what we discussed in the Slack...

File: documentation/staging/content/managing-domains/accessing-the-domain/status-conditions.md [hunk 1]
```diff
@@ -195,15 +195,22 @@ The following is a list of condition types for a Cluster resource.
...
+  - If a cluster has `serverStartPolicy` `Never` or `replicas` `0`,
+    or a cluster is in a domain with `serverStartPolicy` `AdminOnly` or `Never`,
+    then the cluster will have `Available` `False`.
+  - If a cluster and domain both have a `serverStartPolicy` of `IfNeeded`,
+    and `cluster.spec.replicas` is `1`,
+    then the cluster `Available` will be `True` only when its single pod is `ready`. 
+  - If a cluster and domain both have a `serverStartPolicy` of `IfNeeded`,
+    `cluster.spec.replicas` is `4`,
+    and `cluster.spec.maxUnavailable` is `1` (the default),
+    then the cluster `Available` will be `True` only when three are four of its pods are `ready`.
```
Comment (alai8) [inline documentation/staging/content/managing-domains/accessing-the-domain/status-conditions.md:213]: three are four -> three or four

Comment (ankedia) [reply documentation/staging/content/managing-domains/accessing-the-domain/status-conditions.md]: Fixed.

**Comment by tbarnes-us** [comment]
The doc update LGTM.

**Review by rjeberhard — APPROVED** [review]

**Review by alai8 — APPROVED** [review]
LGTM. Please update PR comment with Never serverStartPolicy related changes. Also, please ask Rosemary to review the doc change. Thanks.

**Review by lennyphan — APPROVED** [review]

File: documentation/staging/content/managing-domains/accessing-the-domain/status-conditions.md [hunk 2]
```diff
@@ -195,15 +195,22 @@ The following is a list of condition types for a Cluster resource.
...
-    or if it is not configured, then in the `domain.spec.replicas` field.
-  * Some of the expected server pods for the cluster are temporarily not ready, and the number of
-    `not ready` server pods is less than or equal to `cluster.spec.maxUnavailable` which defaults to `1`.
-- The `status` attribute is also set to `True` when no WebLogic Server pods are running and this
-  is the expected state, such as when `cluster.spec.replicas` is set to `0`, or when
-  `cluster.spec.serverStartPolicy` is set to `Never`.
+- The `status` attribute is set to `True` when a sufficient number of WebLogic Server pods are
+  ready in the cluster. Both of the following must be true:
+  - At least one pod in the cluster is expected to run and is `ready`.
+  - The number of `not ready` server pods which are expected to run
```
Comment (rosemarymarano) [inline documentation/staging/content/managing-domains/accessing-the-domain/status-conditions.md:201]: pods which are expected to run -> pods, which are expected to run, (two commas needed)

File: documentation/staging/content/managing-domains/accessing-the-domain/status-conditions.md [hunk 3]
```diff
@@ -195,15 +195,22 @@ The following is a list of condition types for a Cluster resource.
...
+    is less than or equal to `cluster.spec.maxUnavailable` which defaults to `1`.
```
Comment (rosemarymarano) [inline documentation/staging/content/managing-domains/accessing-the-domain/status-conditions.md:202]: is less than or equal to `cluster.spec.maxUnavailable`, which defaults to `1`. -> (at a minimum, you need a comma before the word "which') 
If you want to make it more explicit, then consider this wording -> is less than or equal to the value of `cluster.spec.maxUnavailable`, which defaults to `1`. **OR** is less than or equal to the value of the `cluster.spec.maxUnavailable` field, which defaults to `1`.

File: documentation/staging/content/managing-domains/accessing-the-domain/status-conditions.md [hunk 4]
```diff
@@ -195,15 +195,22 @@ The following is a list of condition types for a Cluster resource.
...
+- Examples:
+  - If a cluster has `serverStartPolicy` `Never` or `replicas` `0`,
+    or a cluster is in a domain with `serverStartPolicy` `AdminOnly` or `Never`,
+    then the cluster will have `Available` `False`.
```
Comment (rosemarymarano) [inline documentation/staging/content/managing-domains/accessing-the-domain/status-conditions.md:206]: Re-write (I can hardly understand the meaning of this sentence, but here's a stab at it): If a cluster has a `serverStartPolicy` of `Never`, (comma) or `replicas` is `0`, or a cluster is in a domain with a `serverStartPolicy` of `AdminOnly` or `Never`, (comma) then the cluster will be `Available`, `False`.

**Review by rosemarymarano — COMMENTED** [review]
review edits and suggestions

**Pull request merged**